### PR TITLE
gh #248  python l3 hdmiIn enhancement

### PR DESCRIPTION
**Pull request by ishthiyaqahmed**
Hdmi In L3 enhancement.

### CODE REVIEW

File: host/tests/L3_TestCases/dsHdmiIn/dsAudio_L3_Runall_Source.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path))
+sys.path.append(os.path.join(dir_path, "../../"))
+sys.path.append(os.path.join(dir_path, "../../raft"))
+
+from raft.framework.core.logModule import logModule
+
+def Runall_L3():
+    skipTests = ["test10"]
```
Comment (akhilbhas) [inline host/tests/L3_TestCases/dsHdmiIn/dsAudio_L3_Runall_Source.py:37]: Test10 is skipped in both Sink and source RunAll file

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_L3_Runall_Sink.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path))
+sys.path.append(os.path.join(dir_path, "../../"))
+sys.path.append(os.path.join(dir_path, "../../raft"))
+
+from raft.framework.core.logModule import logModule
+
+def Runall_L3():
+    skipTests = ["test10"]
```
Comment (akhilbhas) [inline host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_L3_Runall_Sink.py:37]: Test10 is skipped in both Sink and Source RunAll file

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_test11_GetEDIDInfo_Verify.py [hunk 3]
```diff
@@ -26,9 +26,11 @@
 from enum import Enum, auto
 
 dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path))
```
Comment (akhilbhas) [inline host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_test11_GetEDIDInfo_Verify.py:29]: Why do we need this os.path.join(dir_path) in all files ?

**Review by akhilbhas — COMMENTED** [review]
Please find the review comments.

**Review by Anbukannadhasan — COMMENTED** [review]
Added review comments

File: host/tests/L3_TestCases/dsHdmiIn/dsAudio_L3_Runall_Source.py (hunk 1 above)
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsAudio_L3_Runall_Source.py:37]: no for source device run all not required , will remove the file.

File: host/tests/L3_TestCases/dsHdmiIn/dsAudio_L3_Runall_Source.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+from pathlib import Path
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path))
+sys.path.append(os.path.join(dir_path, "../../"))
+sys.path.append(os.path.join(dir_path, "../../raft"))
+
+from raft.framework.core.logModule import logModule
+
+def Runall_L3():
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsAudio_L3_Runall_Source.py:36]: its only for source device which has Hdmiinput support , currently we don't have that device.

[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsAudio_L3_Runall_Source.py:36]: for source we need not have to run the test cases ,as we don't have source device which support hdmiin ports.

File: host/tests/L3_TestCases/dsHdmiIn/dsAudio_L3_Runall_Source.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+#** *****************************************************************************
+# *
+# * If not stated otherwise in this file or this component's LICENSE file the
+# * following copyright and licenses apply:
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsAudio_L3_Runall_Source.py]: captured the latest changes in docs.

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_L3_Runall_Sink.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+from pathlib import Path
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(dir_path))
+sys.path.append(os.path.join(dir_path, "../../"))
+sys.path.append(os.path.join(dir_path, "../../raft"))
+
+from raft.framework.core.logModule import logModule
+
+def Runall_L3():
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_L3_Runall_Sink.py:36]: its only for source device which has Hdmiinput support , currently we don't have that device.

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py [hunk 7]
```diff
@@ -71,13 +72,14 @@ def __init__(self, testName:str, qcId:str, log:logModule=None ):
 
         # Set up paths and URLs for device test setup
         deviceTestSetup = self.cpe.get("test")
-        #socVendor = self.cpe.get("soc_vendor")
+        socVendor = self.cpe.get("soc_vendor")
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py:75]: yeah its required when exporting runtime variables.

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py [hunk 8]
```diff
@@ -93,15 +95,6 @@ def testDownloadAssets(self):
         # List of streams with path
         self.testStreams = []
 
-        self.deviceDownloadPath = self.cpe.get("target_directory")
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py:96]: its already removed.

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_L3_Runall_Sink.py [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+#** *****************************************************************************
+# *
+# * If not stated otherwise in this file or this component's LICENSE file the
+# * following copyright and licenses apply:
+# *
+# * Copyright 2024 RDK Management
+# *
+# * Licensed under the Apache License, Version 2.0 (the "License");
+# * you may not use this file except in compliance with the License.
+# * You may obtain a copy of the License at
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_L3_Runall_Sink.py:11]: yes tested , generated logs but need to test on RDKE vendor image , that i cannot do here need to take the help of onsite guys.

File: host/tests/dsClasses/dsHdmiIn.py [hunk 10]
```diff
@@ -115,16 +117,30 @@ class dsHdmiInClass():
...
         """
         self.moduleName    = "dsHdmiIn"
-        self.menuConfig    =  os.path.join(dir_path, "dsHdmiIn_test_suite.yml")
+        self.testConfigFile    =  os.path.join(dir_path, "dsHdmiIn_testConfig.yml")
         self.testSuite     = "L3 dsHdmiIn"
-        self.deviceProfile = ConfigRead( deviceProfilePath, self.moduleName)
-        self.utMenu        = UTSuiteNavigatorClass(self.menuConfig, self.moduleName, session)
+
+        # Load configurations for device profile and menu
+        self.moduleConfigProfile = ConfigRead( moduleConfigProfileFile , self.moduleName)
```
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/dsClasses/dsHdmiIn.py:133]: yes for edid info reading and spdinfo reading

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_L3_Runall_Sink.py (hunk 2 above)
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_L3_Runall_Sink.py:37]: its only for source device which has Hdmiinput support , currently we don't have that device.

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (Anbukannadhasan) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py:75]: I am not seeing the variable used in helper class. can you check whether the socVendor variable is really used?

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (Anbukannadhasan) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py:96]: def testDownloadAssets(self):
        """
        Downloads the test artifacts and streams listed in the test setup configuration.

        This function retrieves artifacts and saves them on the DUT (Device Under Test).

        Args:
            None
        """
        # List of streams with path
        self.testStreams = []
        
        why this part of the code is present? Are you using it?

File: host/tests/dsClasses/dsHdmiIn.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (Anbukannadhasan) [reply host/tests/dsClasses/dsHdmiIn.py:133]: moduleConfigProfile is used in any test python file?

**Review by Anbukannadhasan — COMMENTED** [review]
Please address the review comments

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_test11_GetEDIDInfo_Verify.py (hunk 3 above)
Comment (akhilbhas) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiIn_test11_GetEDIDInfo_Verify.py:29]: Here the os.path.join seems redundant as dir_path already has full path.
But functionality wise it is not affected

File: host/tests/dsClasses/dsHdmiIn.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/dsClasses/dsHdmiIn.py]: updated used to read edidinfo and spdinfo

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py:96]: removed , above function.

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py:75]: updated, flag is used to enable vendor specific runtime flags.

File: host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (Anbukannadhasan) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py:96]: Please remove testCleanAssets if is not in use

[the remark this answers is not in the document]
Comment (ishthiyaqahmed) [reply host/tests/L3_TestCases/dsHdmiIn/dsHdmiInHelperClass.py:96]: removed.

**Review by akhilbhas — APPROVED** [review]
Approved

**Review by akhilbhas — DISMISSED** [review]
Approved

**Review by Anbukannadhasan — APPROVED** [review]
Approved

**Review by akhilbhas — APPROVED** [review]
Approved

**Pull request merged**